src/Test/L1/Worker/VariableL1Tests.cs

```diff
@@ -0,0 +1,76 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using Microsoft.TeamFoundation.DistributedTask.WebApi;
+using System.Linq;
+using System.Threading.Tasks;
+using Xunit;
+
+namespace Microsoft.VisualStudio.Services.Agent.Tests.L1.Worker
+{
+    public class VariableL1Tests : L1TestBase
+    {
+        [Fact]
+        [Trait("Level", "L1")]
+        [Trait("Category", "Worker")]
+        public async Task SetVariable_ReadVariable()
+        {
+            // Arrange
+            var message = LoadTemplateMessage();
+            // Remove all tasks
+            message.Steps.Clear();
+            // Add variable setting tasks
+            message.Steps.Add(CreateScriptTask("echo \"##vso[task.setvariable variable=testVar]b\""));
+            message.Steps.Add(CreateScriptTask("echo TestVar=$(testVar)"));
+            message.Variables.Add("testVar", "a");
```

Comment:
I think you're going to need this change once readonly variables gets merged in - https://github.com/microsoft/azure-pipelines-agent/pull/2771/commits/f259a86fcb6f6e550a9d6667630bb64e0438fd79

Reply:
I'll change it here to prevent conflicts